Sources/WalletConnectRelay/WebSocketClient/WebSocketClient.swift

```diff
@@ -0,0 +1,118 @@
+import Foundation
+
+enum WebSocketClientError: Error {
+    case errorWithCode(URLSessionWebSocketTask.CloseCode)
+}
+
+struct WebSocketClientFactory: WebSocketFactory {
+    func create(with url: URL) -> WebSocketConnecting {
+        return WebSocketClient(url: url, logger: ConsoleLogger(loggingLevel: .debug))
+    }
+}
+
+final class WebSocketClient: NSObject, WebSocketConnecting {
```

Comment:
do we still need to conform to WebSocketConnecting  protocol?
I was a starscream requirement
wouldn't it be better to refactor Dispatcher and socket would conform to [this](https://developer.apple.com/documentation/foundation/urlsessionwebsockettask) protocol?